samples/cpp/benchmark/throughput_benchmark/main.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <condition_variable>
+#include <string>
+#include <vector>
+
+// clang-format off
+#include "openvino/openvino.hpp"
+
+#include "samples/args_helper.hpp"
+#include "samples/common.hpp"
+#include "samples/latency_metrics.hpp"
+#include "samples/slog.hpp"
+// clang-format on
+
+using Ms = std::chrono::duration<double, std::ratio<1, 1000>>;
+
+int main(int argc, char* argv[]) {
+    try {
+        slog::info << ov::get_openvino_version() << slog::endl;
+        if (argc != 2) {
+            slog::info << "Usage : " << argv[0] << " <path_to_model>" << slog::endl;
+            return EXIT_FAILURE;
+        }
+        // Optimize for throughput. Best throughput can be reached by
+        // running multiple ov::InferRequest instances asyncronously
+        ov::AnyMap tput{{ov::hint::performance_mode.name(), ov::hint::PerformanceMode::THROUGHPUT}};
+
+        // Create ov::Core and use it to compile a model.
+        // Pick a device by replacing CPU, for example MULTI:CPU(4),GPU(8).
+        // It is possible to set CUMULATIVE_THROUGHPUT as ov::hint::PerformanceMode for AUTO device
+        ov::Core core;
+        ov::CompiledModel compiled_model = core.compile_model(argv[1], "CPU", tput);
+        // Create optimal number of ov::InferRequest instances
+        uint32_t nireq = compiled_model.get_property(ov::optimal_number_of_infer_requests);
+        std::vector<ov::InferRequest> ireqs;
+        for (uint32_t i = 0; i < nireq; ++i) {
+            ireqs.push_back(compiled_model.create_infer_request());
+        }
```

Comment:
Using a bit of modern C++ to make us look like pros
```suggestion
        std::vector<ov::InferRequest> ireqs(nireq);
        std::generate(ireqs.begin(), ireqs.end(), compiled_model.create_infer_request);
```
Requires algorithm header

Reply:
Good idea!

It fails with
```
C:\Users\vzlobin\r\openvino\samples\cpp\benchmark\throughput_benchmark\main.cpp(40,86): error C3867: 'ov::CompiledModel::create_infer_request': non-standard syntax; use
 '&' to create a pointer to member [C:\Users\vzlobin\r\openvino\build\samples\cpp\benchmark\throughput_benchmark\throughput_benchmark.vcxproj]
C:\Users\vzlobin\r\openvino\samples\cpp\benchmark\throughput_benchmark\main.cpp(40,14): error C2672: 'std::generate': no matching overloaded function found [C:\Users\vz 
lobin\r\openvino\build\samples\cpp\benchmark\throughput_benchmark\throughput_benchmark.vcxproj]
```
I wrapped `compiled_model.create_infer_request()` in into a lambda.